internal/server/openvpn.go

```diff
@@ -141,3 +143,30 @@ func (h *openvpnHandler) getPortForwarded(w http.ResponseWriter) {
 		w.WriteHeader(http.StatusInternalServerError)
 	}
 }
+
+func (h *openvpnHandler) setPortForwarded(w http.ResponseWriter, r *http.Request) {
+	var data portsWrapper
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&data); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if len(data.Ports) == 0 {
+		http.Error(w, "no port specified", http.StatusBadRequest)
+		return
+	}
+
+	if err := h.pf.SetPortsForwarded(data.Ports); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Let's log out as warning the error from the function call, and only say `"failed setting forwarded ports"` in the http response (for the sake of not exposing too much details to the http caller who might not control the gluetun instance)

```suggestion
	if err := h.pf.SetPortsForwarded(data.Ports); err != nil {
		h.warner.Warn(fmt.Sprintf("failed setting forwarded ports: %s", err))
		http.Error(w, "failed setting forwarded ports", http.StatusInternalServerError)
```

Reply:
Done with 32a7f1f82152eb523c1726dd73cdbb14f5944733.
Also, I assumed that if we want to limit detail exposure through http responses we should give the json deconding logic just above the same treatment.